lib/available_delivery_location_types.js

```diff
@@ -34,8 +59,63 @@ class AvailableDeliveryLocationTypes {
     })
   }
 
+  static _isUnfamiliarPatronType (patronTypeMapping, patronType) {
+    return util.readJson('data/unfamiliar-patron-type.json')
+      .then((result) => {
+        let unfamiliarPTypeList = result.unfamiliarPatronType
+
+        if (!patronTypeMapping[patronType]) {
+          if (!unfamiliarPTypeList.some((item) => { return item === patronType })) {
+            // Set a limited size to the list
+            if (unfamiliarPTypeList.length < 100) {
+              unfamiliarPTypeList.push(patronType)
+              result.unfamiliarPatronType = unfamiliarPTypeList
+              // Update unfamiliar Patron Type list
+              fsWrite(
+                'data/unfamiliar-patron-type.json',
+                result,
+                () => { logger.info('Add new unfamiliar Patron Type to the list') },
+                process.env.NODE_ENV === 'test'
+              )
+            } else {
+              logger.debug(`Unfamiliar Patron Type list has more than 100 items. Patron Type ${patronType} is not put on the list`)
+            }
+          }
+
+          logger.debug(`Patron Type ${patronType} is not recognizable. Here is the list of the unfamiliar Patron Types we have encountered: ${unfamiliarPTypeList}`)
+
+          return true
+        } else {
+          // Remove the PType from the unfamiliar PType list if it is recognizable now
+          unfamiliarPTypeList.forEach((item) => {
+            if (item === patronType) {
+              unfamiliarPTypeList = unfamiliarPTypeList.filter((item) => item !== patronType)
+            }
+          })
+
+          if (result.unfamiliarPatronType.length !== unfamiliarPTypeList.length) {
+            result.unfamiliarPatronType = unfamiliarPTypeList
+            // Update unfamiliar Patron Type list
+            fsWrite(
+              'data/unfamiliar-patron-type.json',
+              result,
+              () => { logger.info('Remove recognizable Patron Type from the listt') },
+              process.env.NODE_ENV === 'test'
+            )
+          }
+
+          return false
+        }
+      })
+      .catch((e) => {
+        let errorMessage = `Error checking unfamiliar Patron Type list: ${e}`
+        logger.error(errorMessage)
+        throw new Error(errorMessage)
+      })
+  }
 }
 
+// Why line 42 repeats line 10
```

Comment:
I am not sure why 119 and 121 repeat the code we have before. This just serves as a mark I will remove it before merging.